### DISCUSSION TITLE
Best practices for navigation with results -ish without the boilerplate

### DISCUSSION THREAD

**Discussion by zoltish** [square/workflow-kotlin#1204] [Q&A]
Say you have a screen with a button, clicking the button opens an overlay with a result: Yes/No. The state would contain information on whether to show the overlay, for simplicity sake it could just be a boolean. The boolean is then used to conditionally wrap/include the overlay screen somehow. All of this is fine, its very flexible and works great. 

But looking at libraries like circuit, and in general: the way compose usually works (not that its related, I just like the design). Do you think it would be possible to _somehow_ follow the same pattern, where the button click event handler directly calls `overlay.show { result:YesOrNo -> }`?

I realize this goes against some of the principles in workflows, I just keep writing a bunch of boilerplate code for cases like this and.. there has to be a better way? If there is no result, Ive been fine with having a router of sorts that basically wires the navigation to my root workflow, which already has the boilerplate sorted out and can just .. add another overlay, backstack entry, etc.

**Comment by rjrjr** [comment]
Your view code could throw up its own confirmation dialog (or whatever) and bypass `Overlay` entirely. We've certainly done that in places.

**Reply by zoltish** [reply]
Definitely. I do that too! In my case, the overlay is rather complicated and needs to be run with a workflow; theres data to be loaded, logic to be handled, etc.

**Comment by faraz152** [comment]
the standard workflow pattern here is to make the confirmation a child workflow with an output type. parent calls `renderChild` and handles the output to update its own state. no extra boilerplate — the output mechanism is the result mechanism.

if the overlay is simple enough (just yes/no), a `StatelessWorkflow<Props, Output, Nothing>` keeps it minimal. the parent rendering handles whether to show it by conditionally calling `renderChild`.
